### PR TITLE
use reduce instead of forEach.

### PR DESCRIPTION
**Pull request by nicholaswyoung**
To my mind, `reduce()` is greatly preferred when transforming data, rather than just looping over it. What do you think?

### CODE REVIEW

File: examples/real-world/actions/index.js [hunk 1]
```diff
@@ -4,9 +4,10 @@ const SUCCESS = 'SUCCESS'
 const FAILURE = 'FAILURE'
 
 function createRequestTypes(base) {
-  const res = {};
-  [REQUEST, SUCCESS, FAILURE].forEach(type => res[type] = `${base}_${type}`)
-  return res;
+  return [REQUEST, SUCCESS, FAILURE].reduce((acc, type) => {
+		acc[type] = `${base}_${type}`
+		return acc
+	}, {})
```
Comment (Wenqer) [inline examples/real-world/actions/index.js]:
```
return [REQUEST, SUCCESS, FAILURE].reduce((acc, type) => 
  Object.assign(acc, {[type]: `${base}_${type}`}), {})
```

Comment (smashercosmo) [inline examples/real-world/actions/index.js]: That's less performant

Comment (nicholaswyoung) [inline examples/real-world/actions/index.js]: @wenqer `Object.assign` is useful, but mostly for merging objects. Not so much here.

**Comment by yelouafi** [comment]
I think it's not a big deal here to mutate the object. Thanks

**Pull request merged**